x-pack/plugins/secops/public/components/timeline/helpers.tsx

```diff
@@ -4,19 +4,38 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { EventsProps } from '../../containers/events';
+import { isEmpty } from 'lodash/fp';
+import { StaticIndexPattern } from 'ui/index_patterns';
+
+import { convertKueryToElasticSearchQuery } from '../../lib/keury';
 import { DataProvider } from './data_providers/data_provider';
 
 // TODO: implement boolean logic to combine queries instead of just taking the first provider
 export const combineQueries = (
-  dataProviders: DataProvider[]
-): { queryProps: EventsProps; resParm: string } | null =>
-  dataProviders.length
-    ? {
-        queryProps: dataProviders[0].componentQueryProps as EventsProps,
-        resParm: dataProviders[0].componentResultParam,
+  dataProviders: DataProvider[],
+  indexPattern: StaticIndexPattern
+): { filterQuery: string } | null => {
+  if (isEmpty(dataProviders)) {
+    return null;
+  }
+  const globalQuery = dataProviders.reduce((query, dataProvider) => {
+    if (dataProvider.enabled) {
+      if (query !== '') {
+        query += ' or ';
       }
-    : null;
+      query += `(${dataProvider.queryMatch}${
+        dataProvider.queryDate ? ` and ${dataProvider.queryDate})` : ')'
+      }`;
+    }
+    return query;
+  }, '');
+  if (isEmpty(globalQuery)) {
```

Comment:
consider adding an empty line above this one (to make this block a bit less dense)